hpeOneView/resources/settings/appliance_network_interfaces.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+###
+# (C) Copyright [2021] Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from future import standard_library
+
+standard_library.install_aliases()
+
+from hpeOneView.resources.resource import Resource
+
+
+class ApplianceNetworkInterfaces(Resource):
+    """
+    ApplianceNetworkInterface API Client.
+
+    """
+    URI = '/rest/appliance/network-interfaces'
+
+    def __init__(self, connection, data=None):
+        super(ApplianceNetworkInterfaces, self).__init__(connection, data)
+
+    def get_all(self):
+        return super(ApplianceNetworkInterfaces, self).get_by_uri(self.URI)
+
+    def get_by_mac_address(self):
```

Comment:
Also add a method for below endpoint
```GET - /rest/appliance/network-interfaces/mac-addresses```

Reply:
The above method get_by_mac_address implemented for this endpoint only. The other endpoint /rest/appliance/network-interfaces/{macAddresses} uses get_by_id method itself.

Reply:
ok.. I was confused with the function name. I recommend something like ```get_all_mac_address```
Also since the response of ```/rest/appliance/network-interfaces/mac-addresses``` is similar to ```get_all``` response which contains members inside it. Please try to return only the members.